Node.js: fixed "httpVersion" variable format

andrey-zelenkov opened the pull request:
According to the Node.js documenation this variable
should only include numbering scheme.

Thanks to @dbit-xia.

Closes: https://github.com/nginx/unit/issues/1085

ac000 commented on src/nodejs/unit-http/unit.cpp:
```diff
@@ -602,7 +603,12 @@ Unit::create_headers(nxt_unit_request_info_t *req, napi_value request)
-    set_named_property(request, "httpVersion", r->version, r->version_length);
+
+    // trim the "HTTP/" protocol prefix
+    p = (char *) nxt_unit_sptr_get(&r->version);
```
Hi @andrey-zelenkov 

I take it this cast is required due to C++? (In C it wouldn't be needed).

andrey-zelenkov replied:
Yeah, I was also surprised that cast is required here. I was getting following error without it (line number is slightly shifted):

```
npm ERR! ../unit.cpp:611:26: error: invalid conversion from ‘void*’ to ‘char*’ [-fpermissive]
npm ERR!   611 |     p = nxt_unit_sptr_get(&r->version);
npm ERR!       |         ~~~~~~~~~~~~~~~~~^~~~~~~~~~~~~
npm ERR!       |                          |
npm ERR!       |                          void*
```

ac000 replied:
OK.

andrey-zelenkov commented:
Added description in `docs/changes.xml`:

```
git range-diff cee7b11f...80a9311a
1:  cee7b11f ! 1:  80a9311a Node.js: fixed "httpVersion" variable format
    @@ Commit message
     
         Closes: https://github.com/nginx/unit/issues/1085
     
    + ## docs/changes.xml ##
    +@@ docs/changes.xml: with Next.js.
    + </para>
    + </change>
    + 
    ++<change type="bugfix">
    ++<para>
    ++ServerRequest.httpVersion variable format in Node.js module.
    ++</para>
    ++</change>
    ++
    ++
    + </changes>
    + 
    + 
    +
      ## src/nodejs/unit-http/unit.cpp ##
     @@ src/nodejs/unit-http/unit.cpp: Unit::get_server_object()
      void
```
